plugins/woocommerce-blocks/assets/js/base/components/button/index.tsx

```diff
@@ -23,48 +33,76 @@ export interface ButtonProps
 	 * Button variant
 	 */
 	variant?: 'text' | 'contained' | 'outlined';
-	/**
-	 * The URL the button should link to.
-	 */
-	href?: string | undefined;
-}
 
-export interface AnchorProps extends Omit< ButtonProps, 'href' > {
-	/**
-	 * Button href
-	 */
-	href?: string | undefined;
+	removeTextWrap?: boolean;
 }
 
 /**
  * Component that visually renders a button but semantically might be `<button>` or `<a>` depending
  * on the props.
  */
-const Button = ( {
-	className,
-	showSpinner = false,
-	children,
-	variant = 'contained',
-	...props
-}: ButtonProps ): JSX.Element => {
-	const buttonClassName = classNames(
-		'wc-block-components-button',
-		'wp-element-button',
-		className,
-		variant,
-		{
-			'wc-block-components-button--loading': showSpinner,
+const Button = forwardRef< HTMLButtonElement, ButtonProps | LinkProps >(
```

Comment:
I think we need to do 
```suggestion
const Button = forwardRef< HTMLButtonElement, ButtonProps & LinkProps >(
```

here because otherwise it results in TS errors for me: `TS2339: Property removeTextWrap does not exist on type PropsWithChildren<LinkProps | ButtonProps>` however that seems to make `href` a required prop. We could then make href optional since it's not always going to be a link.

Reply:
Hmm, I think I need to find a way around that because I was hoping to avoid an intersection type here to delineate between the 2 usages